auth-web/src/components/auth/common/AccountBusinessType.vue

```diff
@@ -176,151 +181,241 @@
+    const businessTypeCodes = computed(() => codesStore.businessTypeCodes)
+    const isCurrentGovnOrg = currentOrganization.value?.accessType === AccessType.GOVN
 
-  private autoCompleteIsActive: boolean = false
-  private autoCompleteSearchValue: string = ''
-  private isLoading = false
+    const accountInformationForm = ref(null)
+    const orgType = ref(AccountType.INDIVIDUAL)
+    const autoCompleteIsActive = ref(false)
+    const autoCompleteSearchValue = ref('')
+    const isLoading = ref(false)
+    const isBusinessAccount = ref(false)
+    const name = ref('')
+    const businessType = ref('')
+    const businessSize = ref('')
+    const governmentSize = ref('')
+    const branchName = ref('')
+    const isIndividualAccount = ref(false)
+    const isGovnAccount = ref(false)
+    const orgNameRules = ref([])
+    const orgBusinessTypeRules = ref([])
+    const orgBusinessSizeRules = ref([])
 
-  $refs: {
-    accountInformationForm: HTMLFormElement,
-    name: HTMLFormElement,
-    businessType: HTMLFormElement,
-    businessSize: HTMLFormElement,
-    isBusinessAccount: HTMLFormElement
-  }
+    const getOrgNameLabel = computed(() => {
+      if (props.govmAccount) {
+        return 'Ministry Name'
+      } else if (isGovnAccount.value) {
+        return 'Government Agency Name'
+      } else if (isBusinessAccount.value) {
+        return 'Legal Business Name'
+      } else {
+        return 'Account Name'
+      }
+    })
 
-  // input fields
-  private isBusinessAccount = false
-  private name = ''
-  private businessType = ''
-  private businessSize = ''
-  private branchName = ''
+    const getOrgTypeDropdownLabel = computed(() =>
+      orgType.value === AccountType.GOVN ? 'Government Agency Type' : 'Business Type'
+    )
 
-  // Input field rules
-  private readonly orgNameRules = [v => !!v || 'An account name is required']
-  private readonly orgBusinessTypeRules = [v => !!v || 'A business type is required']
-  private readonly orgBusinessSizeRules = [v => !!v || 'A business size is required']
+    const getOrgSizeDropdownLabel = computed(() =>
+      orgType.value === AccountType.GOVN ? 'Government Agency Size' : 'Business Size'
+    )
 
-  /** Emits an update message, so that we can sync with parent */
-  @Emit('update:org-business-type')
-  private emitUpdatedOrgBusinessType () {
-    const orgBusinessType: OrgBusinessType = {
-      name: this.name,
-      isBusinessAccount: this.isBusinessAccount,
-      ...((this.govmAccount || this.isBusinessAccount) && { branchName: this.branchName }),
-      ...(this.isBusinessAccount && { businessType: this.businessType, businessSize: this.businessSize, branchName: this.branchName })
+    function cleanOrgInfo () {
+      name.value = ''
+      branchName.value = ''
+      businessType.value = ''
+      businessSize.value = ''
+      if (accountInformationForm.value) {
+        accountInformationForm.value.resetValidation()
+      }
+      if (isGovnAccount.value) {
```

Comment:
the logic doesn't really change here, it's just the string at the end?

Maybe compute the string, leave the logic alone?

Reply:
![image](https://github.com/bcgov/sbc-auth/assets/3484109/2efd3af4-0b7c-4080-9a5e-45c708a83d3e)

calculate for all 3, just plug them in? Looks cleaner than 6 lines I think